internal/internal_test.go

```diff
@@ -1017,6 +1017,110 @@ func TestConfigWithProtoDescriptor(t *testing.T) {
 	}
 }
 
+func TestCheckAllowObjectDecisionReqHeadersToRemove(t *testing.T) {
+	var req ext_authz.CheckRequest
+	if err := util.Unmarshal([]byte(exampleAllowedRequestParsedPath), &req); err != nil {
+		panic(err)
+	}
+
+	module := `
+		package envoy.authz
+
+		default allow = false
+
+		allow {
+			input.parsed_path = ["my", "test", "path"]
+		}
+
+		headers["x"] = "hello"
+		headers["y"] = "world"
+
+		request_headers_to_remove := ["foo", "bar"]
+
+		result["allowed"] = allow
+		result["headers"] = headers
+		result["request_headers_to_remove"] = request_headers_to_remove`
+
+	server := testAuthzServerWithModule(module, "envoy/authz/result", &testPlugin{}, false)
+	ctx := context.Background()
+	output, err := server.Check(ctx, &req)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if output.Status.Code != int32(code.Code_OK) {
+		t.Fatalf("Expected request to be allowed but got: %v", output)
+	}
+
+	response := output.GetOkResponse()
+	if response == nil {
+		t.Fatal("Expected OkHttpResponse struct but got nil")
+	}
+
+	headers := response.GetHeadersToRemove()
+	if len(headers) != 2 {
+		t.Fatalf("Expected two headers but got %v", len(headers))
```

Comment:
[nit] 🤔 I'd use `t.Fatal[f]` only when the rest of the test body is entirely impossible without this check.  I don't suppose that's the case here.